hikari/api/event_manager.py

```diff
@@ -35,6 +35,8 @@
 if typing.TYPE_CHECKING:
     import types
 
+    from typing_extensions import Self
```

Comment:
could type the returns as `typing_extensions.Self` everywhere but that's a bit too verbose for what it is and looks bad

Reply:
I think it's more readable this way. It also shouldn't be confusing to read anyways.